### PR TITLE
UAF-2202 - Create a file-based remote batch job invoker

### PR DESCRIPTION
**Pull request by sskinner**
This pull request contains the code and configuration needed in order to activate the Batch Invoker process within KFS. Once merged, KFS itself needs no further work for the Batch Invoker to function properly, aside from a file needing to be created on the server's filesystem as: "$BATCH_HOME/control/$HOSTNAME.runnable". Only one host's runnable file may be present at a time, in order to avoid multi-node race conditions that can arise when using a shared file system.

Once the runnable file is present, the Batch Involer watches for special 'run' files on the common filesystem under the same control directory. When a run file is dropped into that control directory by an outside process, the Batch Invoker will schedule the corresponding job under KFS to be executed immediately. Concurrently, the Batch Invoker will write to a status file on the common filesystem, in order to communicate progress to outside processes. Finally, a new shell script project, Rhubarb Farmer (in BitBucket), has been created to facilitate dead-simple batch job interaction from the command line.

### CODE REVIEW

File: kfs-core/src/main/resources/institutional-config.properties [hunk 1]
```diff
@@ -1,3 +1,6 @@
+batchinvoker.directory.towatch=${work.directory}/control
+batchinvoker.directory.history=${work.directory}/control/history
```
Comment (ghost) [inline kfs-core/src/main/resources/institutional-config.properties:2]: How are these config properties used? It looks like these values are set in the Spring file.

Comment (sskinner) [inline kfs-core/src/main/resources/institutional-config.properties:2]: After a hipchat, it is clear that these keys are missing from spring-sys.xml, and so were not being pulled as intended. This has been fixed and pushed, thanks Julie.

**Comment by ghost** [comment]
Looks good! Will be discussing testing plan in our meeting tomorrow, so will defer merging for now.

**Comment by ghost** [comment]
Merging this PR as we have a plan in place for testing.

**Pull request merged**